Detectors/EMCAL/simulation/src/Patches.cxx

```diff
@@ -0,0 +1,306 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file Patches.cxx
+/// \brief Implementation of the EMCAL Patches for the LZEROElectronics
+#include <cstring>
+#include <gsl/span>
+#include <fairlogger/Logger.h> // for LOG
+#include "EMCALSimulation/Patches.h"
+#include "EMCALBase/TriggerMappingV2.h"
+
+using namespace o2::emcal;
+
+//____________________________________________________________________________
+Patches::Patches() : mPatchSize(2), mWhichSide(0), mWhichSuperModuleSize(0)
+{
+  // DEFAULT CONSTRUCTOR
+  mFastOrs.resize(96);
+}
+//____________________________________________________________________________
+Patches::Patches(int patchSize, int whichSide, int whichSuperModuleSize) : mPatchSize(patchSize), mWhichSide(whichSide), mWhichSuperModuleSize(whichSuperModuleSize)
+{
+  // CONSTRUCTOR
+  mFastOrs.resize(96);
+}
+//____________________________________________________________________________
+void Patches::init()
+{
+  mIndexMapPatch.resize(77);
+  for (auto IndexMapPatch : mIndexMapPatch) {
+    (std::get<1>(IndexMapPatch)).resize(mPatchSize * mPatchSize);
+  }
+  mFiredFastOrIndexMapPatch.resize(77);
+  for (auto FiredFastOrIndexMapPatch : mFiredFastOrIndexMapPatch) {
+    (std::get<1>(FiredFastOrIndexMapPatch)).resize(mPatchSize * mPatchSize);
+  }
+  mADCvalues.resize(77);
+  mTimesum.resize(77);
+  mPatchIDSeedFastOrIDs.resize(77);
+  mPreviousTimebinADCvalue.resize(77);
+  mFastOrs.resize(96);
+  for (auto FastOr : mFastOrs) {
+    FastOr.init();
+  }
+  assignSeedModuleToAllPatches();
+  assignModulesToAllPatches();
+}
+//____________________________________________________________________________
+void Patches::clear()
+{
+  mIndexMapPatch.clear();
+  mFiredFastOrIndexMapPatch.clear();
+  mADCvalues.clear();
+  mTimesum.clear();
+  mPatchIDSeedFastOrIDs.clear();
+  mPreviousTimebinADCvalue.clear();
+}
+//____________________________________________________________________________
+void Patches::assignSeedModuleToPatchWithSTUIndexingFullModule(int& patchID)
+{
+  if (mWhichSide == 0) {
+    // A side
+    int rowSeed = patchID / 7;
+    int columnSeed = patchID % 7;
+    int SeedID = rowSeed + patchID;
+    std::get<1>(mPatchIDSeedFastOrIDs[patchID]) = SeedID;
```

Comment:
I think what you are calculating here are "indices" and not "seeds", so please call them indices.